### PR TITLE
[MM-17551] Fixed so that team icon settings displays removal tooltip properly

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
When I changed the `setting_picture` component to have a readable remove button, I missed fixing the styling for the Team Icon use of that component. This PR does that fix.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17551

### CODE REVIEW

**Comment by saturninoabril** [comment]
Looks good to me.  I just notice one thing.  When I opened the section, the tooltip ("Remove this icon") is visible.  Is that expected?  I'm hoping that it will only open up on hover to "x" button.
Screen Shot 2019-08-07 at 1 25 17 AM

Same case for user setting for profile picture.

**Comment by devinbinnie** [comment]
> Looks good to me. I just notice one thing. When I opened the section, the tooltip ("Remove this icon") is visible. Is that expected? I'm hoping that it will only open up on hover to "x" button.
> Screen Shot 2019-08-07 at 1 25 17 AM
> 
> Same case for user setting for profile picture.

Yes that's expected, it's happening because the first thing that is focused when you open the Team Icon settings is the 'x' button, it's so we can use keyboard navigation to both add/remove the icon.

**Comment by asaadmahmood** [comment]
@devinbinnie I know that's what happening (first item being focused), but this also happened on another issue, we do not need the items in a section focused at all times, there should be a prop that allows us not to focus the first element on the section, like this one.

**Comment by devinbinnie** [comment]
> @devinbinnie I know that's what happening (first item being focused), but this also happened on another issue, we do not need the items in a section focused at all times, there should be a prop that allows us not to focus the first element on the section, like this one.

@asaadmahmood But then how will the user navigate via keyboard to this element? This would mean they would have to tab through the entire screen before getting back to the remove button. I think it might make sense to not have the tooltip pop up but I think we need something focused first for keyboard navigation to work in a way that makes sense.

**Comment by asaadmahmood** [comment]
@devinbinnie We can have the container (section itself) be focused when a user opens it.
So have a tabIndex=-1 on the container, and set focus to it in that case.
And then the user can press tab again to jump to the first item.

We would also need to remove the visual focus indicator from the container too.

**Comment by devinbinnie** [comment]
> @devinbinnie We can have the container (section itself) be focused when a user opens it.
> So have a tabIndex=-1 on the container, and set focus to it in that case.
> And then the user can press tab again to jump to the first item.
> 
> We would also need to remove the visual focus indicator from the container too.

@asaadmahmood I suppose that could work, but that should be a separate issue as that is outside the scope of what this bug was intended to fix.

**Comment by asaadmahmood** [comment]
@devinbinnie You're right, what we can do for now is move the `OverlayTrigger` inside the button. That would focus the button, but now show the tooltip. We also do not want the tooltip to show if a user navigates through the keyboard, and instead have a read out.

**Comment by devinbinnie** [comment]
> @devinbinnie You're right, what we can do for now is move the `OverlayTrigger` inside the button. That would focus the button, but now show the tooltip. We also do not want the tooltip to show if a user navigates through the keyboard, and instead have a read out.

@asaadmahmood Agreed, but should still be a separate issue, as it affects other parts of the application (same issue on Profile Picture settings). This fix is only intended to hide the big 'remove this icon' text that was popping up.

**Comment by asaadmahmood** [comment]
I'm fine if we want to take this in, but I am afraid this might bring up another bug issue (that the tooltip appears when you open the section).

**Comment by devinbinnie** [comment]
> I'm fine if we want to take this in, but I am afraid this might bring up another bug issue (that the tooltip appears when you open the section).

@asaadmahmood The tooltip popping up when you open the section is already in master, and has been for a few weeks now. I just didn't think it was an issue until now.

**Comment by asaadmahmood** [comment]
I will defer to @esethna.

**Comment by esethna** [comment]
TLDR, the tooltip should only appear on hover. If it's not a regression then it can be filed as a separate bug.

**Comment by devinbinnie** [comment]
Thanks @esethna and @asaadmahmood. I've created https://mattermost.atlassian.net/browse/MM-17581 to deal with that issue.
